pkgs/by-name/th/thorium/package.nix

```diff
@@ -0,0 +1,46 @@
+{
+  stdenv,
+  fetchurl,
+  undmg,
+  lib,
+}:
+stdenv.mkDerivation (finalAttrs: rec {
+  pname = "thorium";
+  version = "126.0.6478.231";
+
+  src = fetchurl {
+    url = "https://github.com/Alex313031/Thorium-MacOS/releases/download/M${finalAttrs.version}/Thorium_MacOS_ARM64.dmg";
+    hash = "sha256-BBqkNbQ7QjCMfU8yQkiRqV3g9ipjco1XKxR7VYVWhig=";
+  };
+
+  nativeBuildInputs = [ undmg ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    undmg ${src}
+    rm ./Applications
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/Applications
+    cp -r ./Thorium.app $out/Applications
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Chromium fork for Linux, Windows, MacOS, Android, and Raspberry Pi named after radioactive element No. 90.";
+    homepage = "https://thorium.rocks/";
+    license = lib.licenses.bsd3;
+    mainProgram = "thorium";
```

Comment:
Sorry, I'm not sure I understand what you mean. Are you referring to the `mainProgram` attribute?

Reply:
I realize now that since the only output is `$out/Thorium.app` that the `mainProgram` attribute is unnecessary... i think?

> We can create a symbolic link for it during the `installPhase`.
I'm unsure what you mean by this too, could you elaborate? Apologies, I'm somewhat new to nix and nixpkgs.

Reply:
> I realize now that since the only output is `$out/Thorium.app` that the `mainProgram` attribute is unnecessary... i think?

That's not what I mean. It's a good idea to provide `meta.mainProgram` as a single-executable application, but you must have `$out/bin/thorium`. You can see the necessity of `bin/thorium` by running `nix run .#thorium` or by evaluating `lib.getExe pkgs.thorium`.

> > We can create a symbolic link for it during the `installPhase`.
> 
> I'm unsure what you mean by this too, could you elaborate? Apologies, I'm somewhat new to nix and nixpkgs.

A straightforward way to have `bin/thorium` is to create a [symbolic link](https://en.wikipedia.org/wiki/Symbolic_link) at `$out/bin/thorium` and point it to the path of the actual executable. You could do it in the `installPhase`. (We usually do it in `postInstall` if the `installPhase` is not overwritten.)